.github/workflows/test-integration-unreleased.yml

```diff
@@ -0,0 +1,53 @@
+name: Integration with Unreleased OpenSearch
+
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+    branches:
+      - "main"
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        entry:
+          - { opensearch_ref: '1.x' }
```

Comment:
Should we include 2.x as well?

Reply:
Yes that will be adding in the Breaking changes PR, just adding this workflow file here so only 1.x is added.

Reply:
PR https://github.com/opensearch-project/opensearch-go/pull/109/files will be updated once this PR is merged.